conn_go18.go

```diff
@@ -20,8 +22,12 @@ func (c *conn) Ping(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	resp, err := c.doRequest(ctx, req)
-	if err != nil || len(resp) < 7 || string(resp[4:7]) != "Ok." {
+	respBody, err := c.doRequest(ctx, req)
+	if err != nil {
+		return driver.ErrBadConn
+	}
+	resp, err := ioutil.ReadAll(respBody)
+	if err != nil || len(resp) != 4 || !strings.HasPrefix(string(resp), "Ok.") {
```

Comment:
Need to call `c.cancel = nil`   after read whole body

Reply:
Fixed!